platform/android/export/export.cpp

```diff
@@ -2136,6 +2179,40 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 
 		Vector<String> enabled_abis = get_enabled_abis(p_preset);
 
+		String project_icon_path = ProjectSettings::get_singleton()->get("application/config/icon");
+
+		//Prepare images to be resized for the icons. If some image ends up being uninitialized,
+		//the default image from the export template will be used.
+		Ref<Image> launcher_icon_image;
+		Ref<Image> launcher_adaptive_icon_foreground_image;
+		Ref<Image> launcher_adaptive_icon_background_image;
+
+		launcher_icon_image.instance();
+		launcher_adaptive_icon_foreground_image.instance();
+		launcher_adaptive_icon_background_image.instance();
+
+		//Regular icon: user selection -> project icon -> default
+		String path = static_cast<String>(p_preset->get(launcher_icons[0].option_id)).strip_edges();
+		if (path.empty()) {
+			ImageLoader::load_image(project_icon_path, launcher_icon_image);
+		} else if (ImageLoader::load_image(path, launcher_icon_image) != OK) {
+			ImageLoader::load_image(project_icon_path, launcher_icon_image);
+		}
+
+		//Adaptive foreground: user selection -> regular icon (user selection -> project icon -> default)
+		path = static_cast<String>(p_preset->get(launcher_adaptive_icon_foregrounds[0].option_id)).strip_edges();
+		if (path.empty()) {
+			launcher_adaptive_icon_foreground_image = launcher_icon_image;
+		} else if (ImageLoader::load_image(path, launcher_adaptive_icon_foreground_image) != OK) {
+			launcher_adaptive_icon_foreground_image = launcher_icon_image;
```

Comment:
Same comment as above. The two clauses can be combined.

Reply:
Done.